xds/src/main/java/io/grpc/xds/XdsServerBuilder.java

```diff
@@ -83,21 +99,20 @@ public static XdsServerBuilder forPort(int port, ServerCredentials serverCredent
   }
```

Comment:
The serverCredentials need to be wrapped within `forPort()` to inject FilterChainMatchingHandler. This needs to be a separate code path than the XdsServerCredentials, as the user may use different credential types and we'd still need the filter matcher stuff to operate.

Reply:
yea i might have noticed this, looks an existing behavior that needs to be fixed.

Reply:
Previously the filter chain matching was built-in to XdsServerCredentials so it wasn't necessary.